File: src/main/kotlin/com/jackdaw/jdwsequencer/model/MIDIUtils.kt
package com.jackdaw.jdwsequencer.model

fun beatsToMilliSeconds(beat: Double, bpm: Int): Long {
    return (beat * (60.0 / bpm) * 1000).toLong();
}

// TODO: Reversed from function above. Anton suggested it might be Beat = (ms/1000) / (60/bpm)
fun msToBeats(ms: Long, bpm: Int): Double {
    return ((ms.toDouble()/1000) * 60) / bpm
}

File: src/model/midi_utils.rs

pub fn beats_to_milli_seconds(beat: f32, bpm: i32) -> i64 {
    (beat * (60.0 / bpm as f32) * 1000.0) as i64
}

// TODO: Reversed from function above. Anton suggested it might be Beat = (ms/1000) / (60/bpm)
pub fn ms_to_beats(ms: i64, bpm: i32) -> f32 {
    ((ms as f32 / 1000.0) * 60.0) / bpm as f32
}
